src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/ServiceValueResolver.php

```diff
@@ -47,6 +47,8 @@ public function supports(Request $request, ArgumentMetadata $argument)
             $controller = ltrim($controller, '\\');
         }
 
+        $controller = $this->normalizeMethod($controller);
```

Comment:
I you don't mind, I'd like to propose an alternative implementation instead of this method:
```diff
a/src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/ServiceValueResolver.php
+++ b/src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/ServiceValueResolver.php
@@ -47,6 +47,10 @@ final class ServiceValueResolver implements ArgumentValueResolverInterface
             $controller = ltrim($controller, '\\');
         }
 
+        if (!$this->container->has($controller) && false !== $i = strrpos($controller, ':')) {
+            $controller = substr($controller, 0, $i).strtolower(substr($controller, $i));
+        }
+
         return $this->container->has($controller) && $this->container->get($controller)->has($argument->getName());
     }
 
@@ -63,6 +67,11 @@ final class ServiceValueResolver implements ArgumentValueResolverInterface
             $controller = ltrim($controller, '\\');
         }
 
+        if (!$this->container->has($controller)) {
+            $i = strrpos($controller, ':');
+            $controller = substr($controller, 0, $i).strtolower(substr($controller, $i));
+        }
+
         yield $this->container->get($controller)->get($argument->getName());
     }
```

Reply:
Ok, good idea..

Reply:
Updated.